backend/queue/queue.go

```diff
@@ -195,17 +191,27 @@ func (q *Queue) swapLane(ctx context.Context, currentKey string, currentRevision
 	return nil
 }
 
+// getAllBackendIDs gets all currently valid BackendIDs.
+func getAllBackendIDs(ctx context.Context, kv clientv3.KV) ([]string, error) {
+	resp, err := kv.Get(ctx, backendIDKeyPrefix, clientv3.WithPrefix())
+	if err != nil {
+		return nil, err
+	}
+	result := make([]string, 0, len(resp.Kvs))
+	for _, kv := range resp.Kvs {
+		result = append(result, string(kv.Value))
+	}
+	return result, nil
+}
+
 // Enqueue adds a new value to the queue. It returns an error if the context is
 // canceled, the deadline exceeded, or if the client encounters an error.
 func (q *Queue) Enqueue(ctx context.Context, value string) error {
-	resp, err := q.cluster.MemberList(ctx)
+	backendIDs, err := getAllBackendIDs(ctx, q.kv)
```

Comment:
The problem with this approach is that a cold start would require iterating these keys and doing puts on them to warm the caches. While I've been thinking about going down that road, I think I would want it to be done as a consistent approach in several places. Caching is easy to do wrong, and it would require a well thought-out design.

Reply:
I agree, let's keep the design simple for now, and look into caching only if it becomes a performance bottleneck (which I doubt will happen)